test/clj/rems/application/test_commands.clj

```diff
@@ -1011,3 +1011,86 @@
+            application (apply-events application [event])]
+        (is (= {:event/type :application.event/remarked
+                :event/time test-time
+                :event/actor handler-user-id
+                :application/id 123
+                :application/comment "handler's remark"
+                :application/public false}
+               event))))
+    (testing "applicants cannot remark"
+      (is (= {:errors [{:type :forbidden}]}
+             (fail-command application
+                           {:type :application.command/remark
+                            :actor applicant-user-id
+                            :comment ""
+                            :public false}
+                           injections))))
+    (testing "commenter cannot remark before becoming commenter"
+      (is (= {:errors [{:type :forbidden}]}
+             (fail-command application
+                           {:type :application.command/remark
+                            :actor "commenter"
+                            :comment ""
+                            :public false}
+                           injections))))
+    (let [event-1 (ok-command application
+                              {:type :application.command/request-comment
+                               :actor handler-user-id
+                               :commenters ["commenter"]
+                               :comment ""}
+                              injections)
+          application (apply-events application [event-1])
+          event-2 (ok-command application
+                              {:type :application.command/remark
+                               :actor "commenter"
+                               :comment "first remark"
+                               :public false}
+                              injections)
+          application (apply-events application [event-2])]
+      (testing "commenter can remark before"
+        (is (= {:event/type :application.event/remarked
+                :event/time test-time
+                :event/actor "commenter"
+                :application/id 123
+                :application/comment "first remark"
+                :application/public false}
+               event-2))
+        (let [event-1 (ok-command application
+                                  {:type :application.command/comment
+                                   :actor "commenter"
+                                   :comment "..."}
+                                  injections)
+              application (apply-events application [event-1])
+              event-2 (ok-command application
+                                  {:type :application.command/remark
+                                   :actor "commenter"
+                                   :comment "second remark"
+                                   :public false}
+                                  injections)
+              application (apply-events application [event-2])]
+          (testing "and after commenting"
```

Comment:
The same here, I believe this string is written to stdout or somewhere if the test fails, so it will be difficult to extract the needed information from this string alone.

Reply:
Never mind, the author told me that inner testing-strings are concatenated with the outer ones.